assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+                .collect(
+                        toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2,
+                                LinkedHashMap::new));
+
+        int i = 0;
+        for (Map.Entry<Team, Double> entry : sorted.entrySet()) {
+            if (i == 6) {
+                System.out.println("Runner: " + entry.getKey().teamName + " Predicted Score: " + entry.getValue());
+            }
+            if (i == 7) {
+                System.out.println("Winner: " + entry.getKey().teamName + " Predicted Score: " + entry.getValue());
+            } else {
+                i++;
+            }
+        }
+    }
+
+
+    public static void nextGenerationPlayers() {
+        for (Team team : teams) {
+            System.out.println("\n" + team.teamName + ":");
+            for (Player player : team.squad) {
+                if (((player.runs > 100) || (player.wickets > 10)) && (player.matches < 15)) {
+                    player.display();
+                }
+            }
+        }
+    }
+
+    public static void matchFixtures() {
+        int k = 1;
+        for (int i = 0; i < 7; i++) {
+            for (int j = i + 1; j < 8; j++) {
+                Match match = new Match(k, teams.get(i), teams.get(j));
+                matches.add(match);
+                k++;
+            }
+        }
+        ArrayList<Match> reversedMatches = new ArrayList<>();
+        for (Match match : matches) {
+            Match match1 = new Match(k, match.secondTeam, match.firstTeam);
+            reversedMatches.add(match1);
+            k++;
+        }
+        matches.addAll(reversedMatches);
+    }
+
+
+    public static void csvWriter() {
+
+        try (PrintWriter writer = new PrintWriter("Matches.csv")) {
+
+            StringBuilder sb = new StringBuilder();
+            int n = 1;
+            Collections.shuffle(matches);
+            for (Match match : matches) {
+                match.matchCount = n;
+                sb.append(match.matchCount);
+                sb.append(',');
+                sb.append(match.firstTeam.teamName);
+                sb.append(',');
+                sb.append(match.secondTeam.teamName);
+                sb.append(',');
+                sb.append(match.venue.venueName);
+                sb.append('\n');
+                n++;
+            }
+            writer.write(sb.toString());
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+
+    public static void main(String[] args) throws FileNotFoundException {
+
+        for (int i = 0; i < 8; i++) {
+            Team newTeam = new Team(teamNames[i], venueNames[i]);
+            teams.add(newTeam);
+        }
+        Scanner scan = new Scanner(new File("IPL_2021_data.csv"));
+        scan.useDelimiter("\n");
+        scan.next();
+        while (scan.hasNext()) {
+            String line = scan.next();
+            String[] rowInDataset = line.split(",");
```

Comment:
since you know the columns in the file you can store each column value in a variable and have a constructor having 7-8 parameters so it is more clear